packages/data-broker/src/tests/services/weather/WeatherService.test.js

```diff
@@ -0,0 +1,234 @@
+    resetMocks();
+  });
+
+  describe('basic operation', () => {
+    it('returns analytics data when requesting data for dataElements', async () => {
+      const mockModels = await createMockModelsStubWithMockEntity();
+
+      const mockApiResponse = {
+        data: [
+          {
+            precip: 23.6,
+            max_temp: 29.8,
+            min_temp: 24,
+            datetime: '2020-08-20',
+          },
+          {
+            precip: 5,
+            max_temp: 6,
+            min_temp: 7,
+            datetime: '2020-08-21',
+          },
+        ],
+        sources: [],
+      };
+
+      const mockApi = createWeatherApiStub(mockApiResponse);
+
+      const service = new WeatherService(mockModels, mockApi);
+
+      const actual = await service.pull(
+        [
+          {
+            model: {},
+            id: '12345_PRECIP',
+            code: 'PRECIP',
+            type: 'dataElement',
+            service_type: 'weather',
+            config: {},
+          },
+        ],
+        'dataElement',
+        getMockOptionsArg(),
+      );
+
+      expect(actual.results).to.deep.equal([
+        { dataElement: 'PRECIP', value: 23.6, organisationUnit: 'MELB', period: '20200820' },
+        { dataElement: 'PRECIP', value: 5, organisationUnit: 'MELB', period: '20200821' },
+      ]);
+    });
+
+    it('returns events data when requesting data for a dataGroup', async () => {
+      const mockModels = await createMockModelsStubWithMockEntity();
+
+      const mockApiResponse = {
+        data: [
+          {
+            precip: 23.6,
+            max_temp: 29.8,
+            min_temp: 24,
+            datetime: '2020-08-20',
+          },
+          {
+            precip: 5,
+            max_temp: 6,
+            min_temp: 7,
+            datetime: '2020-08-21',
+          },
+        ],
+        sources: [],
+      };
+
+      const mockApi = createWeatherApiStub(mockApiResponse);
+
+      const service = new WeatherService(mockModels, mockApi);
+
+      const actual = await service.pull(
+        [
+          {
+            model: {},
+            id: 'SOME_DATA_GROUP_ID', // data group doesn't matter, it will return the data elements it knows about
+            code: 'SOME_DATA_GROUP_CODE',
+            type: 'dataGroup',
+            service_type: 'weather',
+            config: {},
+          },
+        ],
+        'dataGroup',
+        getMockOptionsArg(),
+      );
+
+      expect(actual.results).to.deep.equal([
+        {
+          organisationUnit: 'MELB',
+          PRECIP: 23.6,
+          MAX_TEMP: 29.8,
+          MIN_TEMP: 24,
+          period: '20200820',
+        },
+        {
+          organisationUnit: 'MELB',
+          PRECIP: 5,
+          MAX_TEMP: 6,
+          MIN_TEMP: 7,
+          period: '20200821',
+        },
+      ]);
+    });
+  });
+
+  describe('gets current weather', () => {
+    it("calls the api with yesterday's date when no dates are provided (simple)", async () => {
```

Comment:
Have you run eslint and prettier? Would expect these to be single quotes

Reply:
Yeah my IDE changed that from `'abc \' def'` to `"abc ' def"`. Does that happen in yours as well? Or I can find how to turn it off.

Reply:
Oh I missed the `'` in there - this is fine!